$NetBSD$

Keep generated output in the shell process.  Grandchildren of GNV Bash do
not reliably inherit an append descriptor on OpenVMS.

--- include/MKkey_defs.sh.orig	2025-06-14 14:44:56.000000000 +0000
+++ include/MKkey_defs.sh
@@ -44,8 +44,8 @@
 pass2=pass2_$$
 pass3=pass3_$$
 pass4=pass4_$$
-trap 'rm -f $data pass[1234]_$$; exit 1' 1 2 3 15
-trap 'rm -f $data pass[1234]_$$' 0
+trap 'rm -f "$data" "$pass1" "$pass2" "$pass3" "$pass4"; exit 1' 1 2 3 15
+trap 'rm -f "$data" "$pass1" "$pass2" "$pass3" "$pass4"' 0
 
 # change repeated tabs (used for readability) to single tabs (needed to make
 # awk see the right field alignment of the corresponding columns):
@@ -63,17 +63,11 @@
 fi
 
 # add keys that we generate automatically:
-cat >>$data <<EOF
-key_resize	kr1	str	R1	KEY_RESIZE	+	NCURSES_SIGWINCH 	Terminal resize event
-EOF
-
-THIS=./`basename "$0"`
-
-cat <<EOF
-/*
- * These definitions were generated by $THIS $*
- */
-EOF
+printf '%s\n' 'key_resize	kr1	str	R1	KEY_RESIZE	+	NCURSES_SIGWINCH 	Terminal resize event' >>"$data"
+
+THIS=./${0##*/}
+
+printf '%s\n' '/*' " * These definitions were generated by $THIS $*" ' */'
 
 # KEY_RESET
 maxkey=345
@@ -187,11 +181,15 @@
 	}
 '
 if test $pass = 1 ; then
-	maxkey=`cat $pass1`
+	IFS= read -r maxkey <"$pass1"
 fi
 
 done
 
-cat $pass2
-cat $pass3
-cat $pass4
+for output in "$pass2" "$pass3" "$pass4"
+do
+	while IFS= read -r line || test -n "$line"
+	do
+		printf '%s\n' "$line"
+	done <"$output"
+done
